front/src/modules/simulationResult/components/SpaceTimeChart/useGetProjectedTrainOperationalPoints.ts

```diff
@@ -88,9 +88,13 @@ const useGetProjectedTrainOperationalPoints = (
         } else {
           // If the manchette hasn't been saved, we want to display by default only
           // the waypoints with CH BV/00/'' and the path steps (origin, destination, vias)
-          operationalPointsWithUniqueIds = operationalPointsWithUniqueIds.filter((op) =>
-            op.extensions?.sncf ? isStation(op.extensions.sncf.ch) || op.weight === 100 : true
-          );
+          operationalPointsWithUniqueIds = operationalPointsWithUniqueIds.filter((op, i) => {
+            if (i === 0 || i === operationalPointsWithUniqueIds.length - 1) return true;
+            // handle waypoints added from the map
+            if (!op.extensions?.sncf) return true;
+            // handle waypoints added from the pathfinding or operational points on path
+            return isStation(op.extensions.sncf.ch) || op.weight === 100;
+          });
```

Comment:
```suggestion
          if (operationalPointsWithUniqueIds.length > 0) {
  const lastIndex = operationalPointsWithUniqueIds.length - 1;

  operationalPointsWithUniqueIds = operationalPointsWithUniqueIds.filter((op, i) => {
    if (i === 0 || i === lastIndex) return true;
    if (!op.extensions?.sncf) return true;
    return isStation(op.extensions.sncf.ch) || op.weight === 100;
  });
}
```

We could maybe use a const like this to avoid recalculation on each iteration of the filter

Reply:
Done